### PR TITLE
"EC2 Workloads at Scale" workshop (ENG). Multiple fixes

### LINKED ISSUE #143: Running EC2 Workloads at Scale lab which prevents Code Deploy from successfully deploying to all EC2 instances.
At this URL: https://ec2spotworkshops.com/running-amazon-ec2-workloads-at-scale/deploy_app_codedeploy.html
Between steps 4 and 5, participants needs to change 
Ec2-spot-workshops\workshops\running-amazon-ec2-workloads-at-scale\koel\composer.json
Line 30 is now case sensitive and the “S” in stream needs to be lower case.
From: "mikey179/vfsStream": "^1.6"
To: "mikey179/vfsstream": "^1.6"
 
This can either be corrected in the lab instructions or by updating the original koel package.

### PR DESCRIPTION
**Pull request by VladimirKuletski**
*Description of changes:*

- [linting] hard tabs to spaces through the document. No trailing quotes in render (noticeable in FF and Chrome)
- [linting] `bash` to code blocks
- `sed` to fix application start issue (Vfs -> vfs) in `composer.json`. Framed as additional step. Please modify description if needed 
Fixes #143 
- [linting] block type change in "Increasing resilience when using Spot Instances" in `spot_resilience.md`
- [linting] json pretty in `spot_resilience.md`

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Comment by ruecarlo** [comment]
Thanks for the contribution. I'll go through the workshop and merge if all goes well :)

**Comment by VladimirKuletski** [comment]
Would be awesome.
PR is the result of ID run for one of the customers.

**Comment by VladimirKuletski** [comment]
Any luck with review?

**Comment by ruecarlo** [comment]
Apologies @VladimirKuletski , we have been revamping other critical content... We are back with updating the workshops and will be picking this one in the next week or so.

File: content/running-amazon-ec2-workloads-at-scale/create_lt.md [hunk 1]
```diff
@@ -59,12 +59,12 @@ Now that the launch template is created, browse to the [Launch Templates console
...
+    ```
 
 * Verify that the contents of the launch template user data are correct:
 
-	```
-	aws ec2 describe-launch-template-versions  --launch-template-name runningAmazonEC2WorkloadsAtScale --output json | jq -r '.LaunchTemplateVersions[].LaunchTemplateData.UserData' | base64 --decode
-	```
\ No newline at end of file
+    ```bash
+    aws ec2 describe-launch-template-versions  --launch-template-name runningAmazonEC2WorkloadsAtScale --output json | jq -r '.LaunchTemplateVersions[]    LaunchTemplateData.UserData' | base64 --decode
```
Comment (jagpk) [inline content/running-amazon-ec2-workloads-at-scale/create_lt.md:69]: revert this change, LaunchTemplateVersions[].LaunchTemplateData.UserData is correct. The extra space you added between .LaunchTemplateVersions[]    LaunchTemplateData.UserData is incorrect.

aws ec2 describe-launch-template-versions  --launch-template-name runningAmazonEC2WorkloadsAtScale --output json | jq -r '.LaunchTemplateVersions[].LaunchTemplateData.UserData' | base64 --decode

File: content/running-amazon-ec2-workloads-at-scale/deploy_app_codedeploy.md [hunk 2]
```diff
@@ -26,93 +27,109 @@ You will now deploy your application to the EC2 instances launched by the Auto S
...
     cd koel && git checkout v3.7.2
     ```
-    
+
 {{% notice note %}}
 You'll get an update about being in 'detached HEAD' state. This is normal.
 {{% /notice %}}
 
-5. Next, copy the CodeDeploy configs into the root level of the koel application directory:
+5. Apply fix to **composer.json**:
```
Comment (jagpk) [inline content/running-amazon-ec2-workloads-at-scale/deploy_app_codedeploy.md:43]: Please change this in composer.json file directly. instead of using sed to vfsStream with vfsstream.

File: content/running-amazon-ec2-workloads-at-scale/deploy_lb.md [hunk 3]
```diff
@@ -11,64 +11,66 @@ Each target group routes requests to one or more registered targets, such as EC2
 
 1. Execute the following command to edit **application-load-balancer.json** and update the values with the resources created by Cloudformation.
 
-	```
-	sed -i.bak -e "s/%publicSubnet1%/$publicSubnet1/g" -e "s/%publicSubnet2%/$publicSubnet2/g" -e "s/%loadBalancerSecurityGroup%/$loadBalancerSecurityGroup/g" application-load-balancer.json
-	```
+    ```bash
+    sed -i.bak -e "s/%publicSubnet1%/$publicSubnet1/g" -e "s/%publicSubnet2%/$publicSubnet2/g" -e "s/%loadBalancerSecurityGroup%/$loadBalancerSecurityGroup/g"    application-load-balancer.json
```
Comment (jagpk) [inline content/running-amazon-ec2-workloads-at-scale/deploy_lb.md:15]: remove the extra  space between loadBalancerSecurityGroup/g"    application-load-balancer.json.

**Review by jagpk — COMMENTED** [review]
Thank you for your contribution. Please go through the review comments and make the suggested changes.

File: content/running-amazon-ec2-workloads-at-scale/deploy_app_codedeploy.md [hunk 4]
```diff
@@ -26,93 +27,109 @@ You will now deploy your application to the EC2 instances launched by the Auto S
...
+
     ```bash
     cd ~/environment/ec2-spot-workshops/workshops/running-amazon-ec2-workloads-at-scale/
     
     git clone https://github.com/phanan/koel.git
     
     cd koel && git checkout v3.7.2
     ```
-    
+
```
[the remark this answers is not in the document]
Comment (VladimirKuletski) [reply content/running-amazon-ec2-workloads-at-scale/deploy_app_codedeploy.md]: Removed. Wasn't breaking sequence in my FF

File: content/running-amazon-ec2-workloads-at-scale/deploy_app_codedeploy.md [hunk 5]
```diff
@@ -26,93 +27,109 @@ You will now deploy your application to the EC2 instances launched by the Auto S
...
 The CodeDeploy console will not default to your current region. Please make sure to click on **Select a Region** in the upper right-hand corner and select your region in the dropdown.
 {{% /notice %}}
 
-8. Next, push the application to the CodeDeploy S3 bucket that the initial CloudFormation template created (which you loaded to the $codeDeployBucket environment variable at the beginning of the workshop):
+9. Next, push the application to the CodeDeploy S3 bucket that the initial CloudFormation template created (which you loaded to the $codeDeployBucket environment variable at the beginning of the workshop):
+
     ```bash
     aws deploy push --application-name koelApp --s3-location s3://$codeDeployBucket/koelApp.zip --no-ignore-hidden-files
     ```
+
```
[the remark this answers is not in the document]
Comment (VladimirKuletski) [reply content/running-amazon-ec2-workloads-at-scale/deploy_app_codedeploy.md]: Removed. Wasn't breaking sequence in my FF

File: content/running-amazon-ec2-workloads-at-scale/deploy_app_codedeploy.md [hunk 6]
```diff
@@ -26,93 +27,109 @@ You will now deploy your application to the EC2 instances launched by the Auto S
...
     ```
 
-13. Take look at the configuration file and then create a deployment running:
-    ```
+14. Take look at the configuration file and then create a deployment running:
+
+    ```bash
     aws deploy create-deployment --cli-input-json file://deployment.json
     ```
+
```
[the remark this answers is not in the document]
Comment (VladimirKuletski) [reply content/running-amazon-ec2-workloads-at-scale/deploy_app_codedeploy.md]: Removed. Wasn't breaking sequence in my FF

File: content/running-amazon-ec2-workloads-at-scale/deploy_lb.md (hunk 3 above)
Comment (VladimirKuletski) [reply content/running-amazon-ec2-workloads-at-scale/deploy_lb.md]: Removed in 47aeaac

File: content/running-amazon-ec2-workloads-at-scale/create_lt.md (hunk 1 above)
Comment (VladimirKuletski) [reply content/running-amazon-ec2-workloads-at-scale/create_lt.md]: Extra space removed. JSON tree structure reverted. 47aeaac

File: content/running-amazon-ec2-workloads-at-scale/deploy_app_codedeploy.md (hunk 2 above)
Comment (VladimirKuletski) [reply content/running-amazon-ec2-workloads-at-scale/deploy_app_codedeploy.md:42]: `composer.json` is stored in different repo and tagged with version.
Details:
In step above we are cloning `https://github.com/phanan/koel.git` repo and switching to `v3.7.2`.

File: content/running-amazon-ec2-workloads-at-scale/spot_resilience.md [hunk 7]
```diff
@@ -13,7 +13,7 @@ By calling the [DetachInstances](https://docs.aws.amazon.com/autoscaling/ec2/API
 
   1. If the Auto Scaling group has a Load Balancer or a Target Group attached (as we have in this workshop), the instance is deregistered from it. Also, if a [deregistration delay](https://docs.aws.amazon.com/elasticloadbalancing/latest/application/load-balancer-target-groups.html#deregistration-delay) is configured for your Load Balancer (or Target Group), the Auto Scaling group waits the configured time to allow in-flight requests to complete (up to the configured timeout, which we have set up to 90 sec).
 
-    You can learn more about the Detaching EC2 Instances from an Auto Scaling group [here](https://docs.aws.amazon.com/autoscaling/ec2/userguide/detach-instance-asg.html).
+  You can learn more about the Detaching EC2 Instances from an Auto Scaling group [here](https://docs.aws.amazon.com/autoscaling/ec2/userguide/detach-instance-asg.html).
```
[the remark this answers is not in the document]
Comment (VladimirKuletski) [reply content/running-amazon-ec2-workloads-at-scale/spot_resilience.md:16]: Can make it somewhere next week. Ether this PR or we can create a separate one.

[the remark this answers is not in the document]
Comment (jagpk) [reply content/running-amazon-ec2-workloads-at-scale/spot_resilience.md:16]: No worries, we can tackle it in a different PR.

**Comment by jagpk** [comment]
Hi @VladimirKuletski, can you pull and rebase once? I will merge the PR post that.

**Comment by VladimirKuletski** [comment]
Done

**Review by jagpk — APPROVED** [review]
Thanks for making the changes. I am approving the PR now.
